insert_data_to_vector_db_weaviate/main.bal

```diff
@@ -0,0 +1,62 @@
+import ballerina/io;
+import ballerinax/googleapis.sheets;
+import ballerinax/openai.embeddings;
+import ballerinax/weaviate;
+
+configurable string openAIToken = ?;
+configurable string weaviateToken = ?;
+configurable string weaviateURL = ?;
+
+configurable string sheetAccessToken = ?;
+configurable string sheetId = ?;
+configurable string sheetName = ?;
+
+const RANGE = "A2:C630";
+const NO_OF_COLUMNS = 3;
+string className = "QuestionAnswerStore";
+
+final embeddings:Client openai = check new ({auth: {token: openAIToken}});
+final weaviate:Client weaviate = check new ({auth: {token: weaviateToken}, timeout: 100}, weaviateURL);
+final sheets:Client gsheets = check new ({auth: {token: sheetAccessToken}});
+
+public function main() returns error? {
+    sheets:Range range = check gsheets->getRange(sheetId, sheetName, RANGE);
+    string[] data = [];
+    weaviate:Object[] documentObjects = [];
+
+    // Iterate through the data stream and extract the content
+    foreach var row in range.values {
+        if (row.length() == NO_OF_COLUMNS) {
+            weaviate:Object obj = {
+                'class: className,
+                properties: {
+                    "title": row[0],
```

Comment:
```suggestion
                    title: row[0],
```

Reply:
This gives a compilation error `invalid key 'title': identifiers cannot be used as rest field keys, expected a string literal or an expression`.

Reply:
@MaryamZi Is this expected?

Reply:
Assuming this is a rest field, yes. This restriction was introduced to minimize room for error with optional fields and open records. https://github.com/ballerina-platform/ballerina-spec/issues/268 has an example for what we try to avoid.